pkg/chains/evm/proto_helpers.go

```diff
@@ -190,141 +258,257 @@ func ConvertCallMsgToProto(msg *evmtypes.CallMsg) (*CallMsg, error) {
+		Addresses:     ConvertAddressesToProto(filter.Addresses),
+		EventSigs:     ConvertHashesToProto(filter.EventSigs),
+		Topic2:        ConvertHashesToProto(filter.Topic2),
+		Topic3:        ConvertHashesToProto(filter.Topic3),
+		Topic4:        ConvertHashesToProto(filter.Topic4),
 		MaxLogsKept:   filter.MaxLogsKept,
 		LogsPerBlock:  filter.LogsPerBlock,
 	}
 }
 
 func ConvertLPFilterFromProto(protoFilter *LPFilter) (evmtypes.LPFilterQuery, error) {
 	if protoFilter == nil {
-		return evmtypes.LPFilterQuery{}, errEmptyFilter
+		return evmtypes.LPFilterQuery{}, ErrEmptyFilter
+	}
+
+	var addresses []evmtypes.Address
+	for i, protoAddress := range protoFilter.GetAddresses() {
+		address, err := ConvertOptionalAddressFromProto(protoAddress)
+		if err != nil {
+			return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert address[%d]: %w", i, err)
+		}
+		addresses = append(addresses, address)
+	}
+
+	sigs, err := ConvertHashesFromProto(protoFilter.GetEventSigs())
+	if err != nil {
+		return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert event sigs: %w", err)
+	}
+
+	t2, err := ConvertHashesFromProto(protoFilter.GetTopic2())
+	if err != nil {
+		return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert topic2: %w", err)
+	}
+
+	t3, err := ConvertHashesFromProto(protoFilter.GetTopic3())
+	if err != nil {
+		return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert topic3: %w", err)
+	}
+
+	t4, err := ConvertHashesFromProto(protoFilter.GetTopic4())
+	if err != nil {
+		return evmtypes.LPFilterQuery{}, fmt.Errorf("failed to convert topic4: %w", err)
 	}
 
 	return evmtypes.LPFilterQuery{
 		Name:         protoFilter.GetName(),
 		Retention:    time.Duration(protoFilter.GetRetentionTime()),
-		Addresses:    ConvertAddressesFromProto(protoFilter.GetAddresses()),
-		EventSigs:    ConvertHashesFromProto(protoFilter.GetEventSigs()),
-		Topic2:       ConvertHashesFromProto(protoFilter.GetTopic2()),
-		Topic3:       ConvertHashesFromProto(protoFilter.GetTopic3()),
-		Topic4:       ConvertHashesFromProto(protoFilter.GetTopic4()),
+		Addresses:    addresses,
+		EventSigs:    sigs,
+		Topic2:       t2,
+		Topic3:       t3,
+		Topic4:       t4,
 		MaxLogsKept:  protoFilter.GetMaxLogsKept(),
 		LogsPerBlock: protoFilter.GetLogsPerBlock(),
 	}, nil
 }
 
-func ConvertFilterToProto(filter evmtypes.FilterQuery) *FilterQuery {
+var ErrTopicsConversion = errors.New("failed to convert topics")
+
+func ConvertFilterToProto(filter evmtypes.FilterQuery) (*FilterQuery, error) {
+	topics, err := convertTopicsToProto(filter.Topics)
+	if err != nil {
+		return nil, errors.Join(ErrTopicsConversion, err)
+	}
+
 	return &FilterQuery{
 		BlockHash: filter.BlockHash[:],
 		FromBlock: valuespb.NewBigIntFromInt(filter.FromBlock),
 		ToBlock:   valuespb.NewBigIntFromInt(filter.ToBlock),
-		Addresses: convertAddressesToProto(filter.Addresses),
-		Topics:    convertTopicsToProto(filter.Topics),
-	}
+		Addresses: ConvertAddressesToProto(filter.Addresses),
```

Comment:
This does not seem to be consistent with ConvertLPFilterFromProto. Shouldn't we always allow an empty address? In case a chain produces logs from it?